gcexport.py

```diff
@@ -443,7 +443,7 @@ def paceOrSpeedFormatted(typeId, parentTypeId, mps):
 		typeId = 4 if absentOrNull('activityType', a) else a['activityType']['typeId']
 
 		startTimeWithOffset = offsetDateTime(a['startTimeLocal'], a['startTimeGMT'])
-		elapsedDuration = details['summaryDTO']['elapsedDuration'] if details['summaryDTO'] else None
+		elapsedDuration = details['summaryDTO']['elapsedDuration'] if 'summaryDTO' in details and 'elapsedDuration' in details['summaryDTO'] else None
```

Comment:
Isn't the first condition in the if statement redundant? If the second condition is true the first condition will also be true.

Reply:
If the first condition was false, the evaluation of the second might throw an exception, might it not? But it is redundant because earlier I check that summaryDTO is present and redownload the JSON otherwise.

I have a private branch where I convert the whole script into a main function and want to extract the functions dealing with JSON and CSV to be able to test them independently with pytest. I'm not quite there yet to have a test for this case; when I'll have the tests I'll confidently refactor to make the code simpler.

Reply:
But, the previous statement, as in row 446 without my change threw an exception for my data. But that might very well be because of missing 'elapsedDuration' rather than 'summaryDTO'.
I thought it prudent to test for both because previous statement tested for summaryDTO, but in an incorrect way I guess.

Reply:
Yes, a missing `elapsedDuration` would do that. And yes, already the previous test on `summaryDTO` was redundant, I missed it when I cleaned up after refactoring.

Thank you for providing a fix!